### PR TITLE
Add Header / Layout components

### PR DESCRIPTION
**Pull request by cdun**
# Purpose of PR

Introduces alpha versions of `Header` and `Layout` components

## PR Checklist

- [x] I have read the relevant `readme.md` file(s)
- [x] All commits follow our [Git commit message convention](https://github.com/contentful/forma-36/tree/main/packages/forma-36-react-components#commits)
- [x] Tests are added/updated/not required
- [x] Tests are passing
- [x] Storybook stories are added/updated/not required
- [x] Usage notes are added/updated/not required
- [x] Has been tested based on [Contentful's browser support](https://www.contentful.com/faq/about-contentful/#which-browsers-does-contentful-support)
- [x] Doesn't contain any sensitive information

### CODE REVIEW

**Review by mshaaban0 — APPROVED** [review]
Nicely done ✅

File: packages/components/header/README.mdx [hunk 1]
```diff
@@ -0,0 +1,51 @@
+---
+title: 'Header'
+type: 'component'
```
Comment (massao) [inline packages/components/header/README.mdx:3]: Maybe they should be on the Layout section??
```suggestion
type: 'component'
section: 'layoutComponents'
```

File: packages/components/layout/examples/BasicLayoutExample.tsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  const [submitting, setSubmitting] = useState(false);
+  const submitForm = () => {
+    setSubmitting(true);
+    setTimeout(() => setSubmitting(false), 1000);
+  };
+
+  return (
+    <Layout
+      header={
+        <Layout.Header>
```
Comment (massao) [inline packages/components/layout/examples/BasicLayoutExample.tsx:24]: We import from Header package on line 10, but still use Layout.Header here

File: packages/website/components/Playground/SandpackRenderer.tsx [hunk 3]
```diff
@@ -96,7 +96,9 @@ export function SandpackRenderer({
           'react-scripts': '^4.0.0',
           '@contentful/f36-components': '^4.0.0',
           '@contentful/f36-avatar': '^4.0.0-alpha.0', // Remove when added to f36-components
+          '@contentful/f36-header': '^4.0.0-alpha.0', // Remove when added to f36-components
           '@contentful/f36-image': '^4.0.0-alpha.0', // Remove when added to f36-components
+          '@contentful/f36-layout': '^4.0.0-alpha.0', // Remove when added to f36-components
```
Comment (massao) [inline packages/website/components/Playground/SandpackRenderer.tsx:101]: We need to start moving things to the f36-components... 😅

File: packages/components/layout/README.mdx [hunk 4]
```diff
@@ -0,0 +1,31 @@
+---
+title: 'Layout'
+type: 'component'
```
Comment (massao) [inline packages/components/layout/README.mdx:3]: Same about the section here

**Review by massao — APPROVED** [review]
LGTM, with some non blockers comments

File: packages/components/layout/examples/BasicLayoutExample.tsx (hunk 2 above)
Comment (cdun) [reply packages/components/layout/examples/BasicLayoutExample.tsx:24]: This is intentional 👍

File: packages/components/header/README.mdx (hunk 1 above)
Comment (cdun) [reply packages/components/header/README.mdx]: Makes sense - updating

**Pull request merged**